actions/lib/base.py

```diff
@@ -11,17 +11,27 @@ def __init__(self, config):
         self._init_dd()
 
     def _init_dd(self):
-        options = {
-            'api_key': self.config['api_key'],
-            'app_key': self.config['app_key']
-        }
+        options = {}
+
+        if 'api_key' in self.config and 'app_key' in self.config:
```

Comment:
You are using the same code as below, this function is not useful anymore.

You can replace it with this kind of code: (and remove the self._init_dd() in __init__())

`
def run(self, **kwargs):
    options = {
        'api_key' = kwargs.get('api_key', self.config['api_key'])).
        'app_key' = kwargs.get('api_key', self.config['app_key'])).
    }
    initialize(**options)
.... blabla ....
`

And now it covers both cases. Maybe if we have no kwargs and no api key we should raise a better exception... instead of the KeyError..